core/src/main/scala/cats/instances/seq.scala

```diff
@@ -17,6 +20,17 @@ trait SeqInstances extends cats.kernel.instances.SeqInstances {
 
       def combineK[A](x: Seq[A], y: Seq[A]): Seq[A] = x ++ y
 
+      override def combineAllOptionK[A](as: IterableOnce[Seq[A]]): Option[Seq[A]] = {
+        val iter = as.iterator
+        if (iter.isEmpty) None else Some(appendAll(iter, Seq.newBuilder[A]).result())
+      }
+
+      override def fromIterableOnce[A](as: IterableOnce[A]): Seq[A] = {
```

Comment:
same pondering about `as.isEmpty`

Reply:
AFAICT `isEmpty` is [deprecated](https://www.scala-lang.org/api/2.13.7/scala/collection/IterableOnce.html#isEmpty:Boolean) in favour of `iterator.isEmpty`

Reply:
otherwise this would be a nice optimization

Reply:
or is allocating an `Iterator` to maybe not have to allocate a builder preferable?